packages/react-reconciler/src/ReactFiberThrow.js

```diff
@@ -466,24 +473,22 @@ function throwException(
                     retryQueue.add(wakeable);
                   }
                 }
+
+                // We only attach ping listeners in concurrent mode. Legacy
+                // Suspense always commits fallbacks synchronously, so there are
+                // no pings.
+                if (suspenseBoundary.mode & ConcurrentMode) {
+                  attachPingListener(root, wakeable, rootRenderLanes);
+                }
               }
-              break;
+              return;
             }
-            // Fall through
           }
-          default: {
-            throw new Error(
-              `Unexpected Suspense handler tag (${suspenseBoundary.tag}). This ` +
-                'is a bug in React.',
-            );
-          }
-        }
-        // We only attach ping listeners in concurrent mode. Legacy Suspense always
```

Comment:
I moved this into the branches above to avoid attaching a redundant listener to the "noop" thenable. That way we can treat any listener that does get attached as a mistake and warn in dev.